manager/integration/tests/test_basic.py

```diff
@@ -4558,3 +4561,66 @@ def test_backup_failed_disable_auto_cleanup(set_random_backupstore,  # NOQA
                       + volume_name + " is deleted"
     except AssertionError:
         pass
+
+
+def test_backup_save_access_mode(client, core_api, apps_api,  # NOQA
+                                 make_deployment_with_pvc):  # NOQA
+    """
+    Test the backup should be save access mode and restore based on the value
+
+    1. Create a volume with 'accessMode' rwx.
+    2. Create the backup for the volume.
+    3. Restore the backup without specific access mode
+    4. Check the restored volume's access mode should be 'readwritemany'
+    5. Restoring the backup using the `readwriteonce` access mode
+    6. Check the restored volume's access mode should be 'readwriteonce'
+    """
+
+    test_volume_name = "test-rwx-volume"
+    test_pv_name = test_volume_name + "-pv"
+    test_pvc_name = test_volume_name + "-pvc"
+    volume_access_mode = "rwx"
+
+    # create a volume
+    client.create_volume(name=test_volume_name, size=str(1 * Gi),
+                         numberOfReplicas=3, accessMode=volume_access_mode)
+    wait_for_volume_creation(client, test_volume_name)
+    volume = wait_for_volume_detached(client, test_volume_name)
+
+    # create the deployment with specific PV & PVC
+    create_pv_for_volume(client, core_api, volume, test_pv_name)
+    create_pvc_for_volume(client, core_api, volume, test_pvc_name)
+    deployment = make_deployment_with_pvc('deployment-rwx-volume-test',
+                                          test_pvc_name, replicas=2)
+
+    # deploy and get a rwx volume
+    apps_api = get_apps_api_client()
+    create_and_wait_deployment(apps_api, deployment)
+    volume.attach(hostId=get_self_host_id())
```

Comment:
You don't need to create a deployment for this test, but rather have a volume w/ access mode, then restore to check the access mode correct.

Also, don't need to call `volume.attach` if you have a deployment w/ the volume, because it will be automatically attached after the deployment deployed.

Reply:
After consult with @chriscchien and the other test case, I will user `create_volume()` function with access mode argument.

https://github.com/longhorn/longhorn-tests/blob/c0f5ac9ca5589e641830fe791497c56494412de1/manager/integration/tests/test_basic.py#L4583-L4593